examples/server/daily-bot-manager.py

```diff
@@ -1,101 +1,165 @@
+        json={
+            'properties': token_props})
+    if res.status_code != 200:
+        if res.status_code != 200:
+            raise Exception(f"Unable to create meeting token: {res.text}")
+
+    meeting_token = res.json()['token']
+    return meeting_token
+
+
+def start_bot(*, bot_path, room_url, token, bot_args, wait_for_bot):
 
+    room_name = get_room_name(room_url)
     proc = subprocess.Popen(
-        [f"python {bot_path} -u {room_url} -t {meeting_token} -k {daily_api_key} {extra_args}"],
+        [f"python {bot_path} -u {room_url} -t {token} -k {daily_api_key} {bot_args}"],
         shell=True,
         bufsize=1,
     )
 
-    # Don't return until the bot has joined the room, but wait for at most 2
-    # seconds.
-    attempts = 0
-    while attempts < 20:
-        time.sleep(0.1)
-        attempts += 1
-        res = requests.get(
-            f"{api_path}/rooms/{room_name}/get-session-data",
-            headers={"Authorization": f"Bearer {daily_api_key}"},
-        )
-        if res.status_code == 200:
-            break
-    print(f"Took {attempts} attempts to join room {room_name}")
-
-    # Additional client config
-    config = {}
-    if os.getenv("CLIENT_VAD_TIMEOUT_SEC"):
-        config['vad_timeout_sec'] = float(
-            os.getenv("DAILY_CLIENT_VAD_TIMEOUT_SEC"))
+    if wait_for_bot:
+        # Don't return until the bot has joined the room, but wait for at most 5
+        # seconds.
+        attempts = 0
+        while attempts < 50:
+            time.sleep(0.1)
+            attempts += 1
+            res = requests.get(
+                f"{api_path}/rooms/{room_name}/get-session-data",
+                headers={"Authorization": f"Bearer {daily_api_key}"},
+            )
+            if res.status_code == 200:
+                print(f"Took {attempts} attempts to join room {room_name}")
+                return True
+
+        # If we don't break from the loop, that means we never found the bot in the room
+        raise Exception("The bot was unable to join the room. Please try again.")
+
+    return True
+
+
+@app.route("/start/<string:text>", methods=["GET", "POST"])
+def start(text):
+    if text in APPS:
+        try:
+            bot_path = APPS[text]
+
+            props = {
+                "room_url": None,
+                "room_properties": None,
+                "token_properties": None,
+                "bot_args": None,
+                "wait_for_bot": True,
+                "duration": None,
+                "redirect": True
+            }
+            props |= request.values.to_dict()  # gets URL params as well as plaintext POST body
+            try:
+                props |= request.json
+            except BaseException:
+                pass
+            if props['redirect'] == "false":
+                props['redirect'] = False
+            if props['wait_for_bot'] == "false":
+                props['wait_for_bot'] = False
+
+            duration = int(os.getenv("DAILY_ROOM_TIMEOUT")
```

Comment:
since we're doing this refactor, might as well just settle on one env variable (either of these is fine) instead of supporting two.